File: Backend/API/Models/Restaurants.Models/Data/OpenHours.cs
using System;
using Models.Restaurants.Models.Enums;

namespace Models.Restaurants.Models.Data
{
    public class OpenHours
    {
        public DateTime Open { get; set; }
        public DateTime Close { get; set; }
        public WeekDay WeekDay { get; set; }

        public Restaurant Restaurant { get; set; }
        public int RestaurantId { get; set; }
    }
}

File: reservation_app/lib/Models/restaurants/open_hours.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:reservation_app/Models/interfaces/json_format.dart';

part 'open_hours.g.dart';

@JsonSerializable()
class OpenHours implements JsonFormat {
  OpenHours(this.open, this.close, this.weekDay);

  final DateTime open;
  final DateTime close;
  int weekDay;

  factory OpenHours.fromJson(Map<String, dynamic> json) => _$OpenHoursFromJson(json);

  Map<String, dynamic> toJson() => _$OpenHoursToJson(this);
}
